cmd/k3s/main.go

```diff
@@ -100,6 +101,30 @@ func findDataDir() string {
 	return dataDir
 }
 
+// findPreferBundledBin searches for prefer-bundled-bin from the CLI args and config file.
+func findPreferBundledBin(args []string) bool {
+	preferRes := configfilearg.MustFindString(args, "prefer-bundled-bin")
+	for i, arg := range args {
+		flagName := "--prefer-bundled-bin"
+		if flagName == arg {
+			if len(args) > i+1 {
+				if res, err := strconv.ParseBool(args[i+1]); err == nil {
+					return res
+				}
```

Comment:
Can't we just set preferRes and then handle actually parsing it at the end, same as if it was passed in the same arg?
```suggestion
				preferRes = args[i+1]
```

Reply:
I have combined both. We still need to use ParseBool to handle cases around the next flag.